# IMPORTS

In [15]:
# importing all the imports needed for this phase
# data manipulation
import pandas as pd
import numpy as np

# visuals
import matplotlib.pyplot as plt
import seaborn as sns

# personal import
import created_functions as cf
import env

# sk imports
from sklearn.model_selection import train_test_split

# ACQUIRE

In [16]:
# getting data from SQL
df =cf.get_telco_data()

this file exists, reading csv


In [17]:
df.head()

payment_type_id  internet_service_type_id  contract_type_id  \
Unnamed: 0                                                                
0                         2                         1                 2   
1                         2                         1                 1   
2                         1                         2                 1   
3                         1                         2                 1   
4                         2                         2                 1   

           customer_id  gender  senior_citizen partner dependents  tenure  \
Unnamed: 0                                                                  
0           0002-ORFBO  Female               0     Yes        Yes       9   
1           0003-MKNFE    Male               0      No         No       9   
2           0004-TLHLJ    Male               0      No         No       4   
3           0011-IGKFF    Male               1     Yes         No      13   
4           0013-EXCHZ  Female               1     Yes         No       3   

           phone_service  ... tech_support streaming_tv streaming_movies  \
Unnamed: 0                ...                                              
0                    Yes  ...          Yes          Yes               No   
1                    Yes  ...           No           No              Yes   
2                    Yes  ...           No           No               No   
3                    Yes  ...           No          Yes              Yes   
4                    Yes  ...          Yes          Yes               No   

           paperless_billing monthly_charges total_charges churn  \
Unnamed: 0                                                         
0                        Yes            65.6         593.3    No   
1                         No            59.9         542.4    No   
2                        Yes            73.9        280.85   Yes   
3                        Yes            98.0       1237.85   Yes   
4                        Yes            83.9         267.4   Yes   

             contract_type  internet_service_type      payment_type  
Unnamed: 0                                                           
0                 One year                    DSL      Mailed check  
1           Month-to-month                    DSL      Mailed check  
2           Month-to-month            Fiber optic  Electronic check  
3           Month-to-month            Fiber optic  Electronic check  
4           Month-to-month            Fiber optic      Mailed check  

[5 rows x 24 columns]

In [18]:
df.shape

(7043, 24)

>there are 7043 rows and 24 columns

important
> gender
> tenure
> senior
> monthly

unimportant
> payment_type_id
> internet_service_type
> contract_type_id
> customer_id ( theres already an index)  

# PREPARE

In [19]:
df = cf.prep_telco(df)

In [20]:
df.head()

gender  senior_citizen partner dependents  tenure phone_service  \
Unnamed: 0                                                                    
0           Female               0     Yes        Yes       9           Yes   
1             Male               0      No         No       9           Yes   
2             Male               0      No         No       4           Yes   
3             Male               1     Yes         No      13           Yes   
4           Female               1     Yes         No       3           Yes   

           multiple_lines online_security online_backup device_protection  \
Unnamed: 0                                                                  
0                      No              No           Yes                No   
1                     Yes              No            No                No   
2                      No              No            No               Yes   
3                      No              No           Yes               Yes   
4                      No              No            No                No   

           tech_support streaming_tv streaming_movies paperless_billing  \
Unnamed: 0                                                                
0                   Yes          Yes               No               Yes   
1                    No           No              Yes                No   
2                    No           No               No               Yes   
3                    No          Yes              Yes               Yes   
4                   Yes          Yes               No               Yes   

            monthly_charges total_charges churn   contract_type  \
Unnamed: 0                                                        
0                      65.6         593.3    No        One year   
1                      59.9         542.4    No  Month-to-month   
2                      73.9        280.85   Yes  Month-to-month   
3                      98.0       1237.85   Yes  Month-to-month   
4                      83.9         267.4   Yes  Month-to-month   

           internet_service_type      payment_type  
Unnamed: 0                                          
0                            DSL      Mailed check  
1                            DSL      Mailed check  
2                    Fiber optic  Electronic check  
3                    Fiber optic  Electronic check  
4                    Fiber optic      Mailed check

### changes to data
this  function takes in the telco database and cleans it
#### deletes
    payment type id
    internet service id
    contract type id
    customer id
turns total charges into a float by deleting empty spaces

In [26]:
#combining the add on columns into 1
#adding 1 for addons and 0 for no addons
df['has_add_ons'] = df[['online_security'
    , 'online_backup', 'device_protection', 'tech_support']].apply(
    lambda row: 1 if any(x == 'Yes' for x in row) else 0, 
    axis=1
)

### data splitting

In [27]:
# splitting the data
train, validate, test = cf.splitting_data(df, 'churn')

In [28]:
# checking out new variable
train.shape, validate.shape, test.shape

((4219, 21), (1406, 21), (1407, 21))

## PREPROCESS
